### ISSUE TITLE
Issue with MultiUrlPicker and uSync

### ISSUE THREAD

**Issue by steadygo** [KevinJump/uSync8#400]
**Description**

A documentType comprises a Multi URL Picker - when the content is synced the template fails to render with error: 

Error Rendering Module "System.NullReferenceException: Object reference not set to an instance of an object. at Umbraco.Cms.Core.PropertyEditors.ValueConverters.MultiUrlPickerValueConverter.ConvertIntermediateToObject(IPublishedElement owner, IPublishedPropertyType propertyType, PropertyCacheLevel referenceCacheLevel, Object inter, Boolean preview) at Umbraco.Cms.Core.Models.PublishedContent.PublishedPropertyType.ConvertInterToObject(IPublishedElement owner, PropertyCacheLevel referenceCacheLevel, Object inter, Boolean preview) at Umbraco.Cms.Core.PublishedCache.PublishedElementPropertyBase.GetValue(String culture, String segment) at Umbraco.Extensions.PublishedPropertyExtension.Value[T](IPublishedProperty property, IPublishedValueFallback publishedValueFallback, String culture, String segment, Fallback fallback, T defaultValue) at Umbraco.Extensions.PublishedElementExtensions.Value[T](IPublishedElement content, IPublishedValueFallback publishedValueFallback, String alias, String culture, String segment, Fallback fallback, T defaultValue) 

The fail is where it tries to get any property of a link.

I am using the following to try to safely get at the link - I was using a strongly typed model but reverted to this from Umbraco Docs for test purposes.

var link = Model.Value<Link>("Link");
if (link != null){}

On re-publishing the node on the destination site the error disappeasrs.

**To Reproduce**
1. Add docType with Multi URL Picker
2. Populate or not - doesn't matter if it is used or not
3. Sync content based upon the docType from one environment to another
4. Render fails

**Expected behavior**
Expected that the property will be readable and not null without having to republish on destination environment

**About your Site (please complete the following information):**
 - Umbraco Version: 9.3.1
 - uSync Version: v9.4.0 + Snapshots (v9.4.0), Exporter (v9.4.0), PeopleEdition (v9.4.0), Publisher (v9.4.0), Complete (v9.4.0)
 - Browser [e.g. chrome, safari] - All Browsers

**Comment by KevinJump** [comment]
Hi, 

I am struggling to reproduce this one locally. 

Just to confirm i have done the following: 
1. Two Sites (Umbraco 9.3.1 / uSync,Complete 9.4.0)
2. Create a new Doctype,  with a MultiURL Picker setup 
(I have used the default setup) 
![image](https://user-images.githubusercontent.com/431231/177347309-38eda34e-7d5e-40a9-be9a-dde3fc2afb24.png)

3. on the doctype 
![image](https://user-images.githubusercontent.com/431231/177347399-573a928f-911b-4f8d-806b-09cdb272e271.png)

4. Then on the source site : I have picked a couple of content items. 

![image](https://user-images.githubusercontent.com/431231/177347545-dadced9d-562b-4b1a-bf4f-17447e8d9cd3.png)

5. really simple doctype for testing...
```html
@using Umbraco.Cms.Core.Models
@using Umbraco.Cms.Web.Common.PublishedModels;
@inherits Umbraco.Cms.Web.Common.Views.UmbracoViewPage
@{
	Layout = "Master.cshtml";
}

<h1>@Model.Value("pageTitle")</h1>

@{
	var linkRaw = Model.Value("primaryLink");
	if (linkRaw != null) { 
		<h4>Not null</h4>
	 }

	var links = Model.Value<List<Link>>("primaryLink");
	if (links != null) {
		
		<ul>
		@foreach(var link in links) {
			<li>@link.Url (@link.Udi)</li>
		}
		</ul>
	}
}
```

I then pushed all the content (homepage and childpage) to the other site. 

and the page renders. 

![image](https://user-images.githubusercontent.com/431231/177348314-daab848c-32a2-4190-a1f8-f521ccc6f84a.png)


---

If there is anything you think i should/could change to get closer to your setup, i will take a look.

**Comment by steadygo** [comment]
Hi Kevin

Thanks for the amazingly quick response.

Looking at my doc type it actually stems from a nested content item – sorry – it’s a bit more sophisticated than initially described.

It consistently fails to sync.

We have a Tab Content Module

![1](https://user-images.githubusercontent.com/4246468/177367281-05b0a5eb-4a48-47a3-8464-eacedfb309db.png)

That comprises a nested content property of element type Tab

![2](https://user-images.githubusercontent.com/4246468/177367332-4c7b1e43-46ca-46e9-9d8b-bf386fe16f72.png)

The Tab Content Type is:

![3](https://user-images.githubusercontent.com/4246468/177367378-34356730-f60b-47af-b55c-d6777a50e822.png)

And the configuration for the Picker is:

![4](https://user-images.githubusercontent.com/4246468/177367401-efdc2cbb-8038-4795-8e78-93b14c933d74.png)

Many Thanks

Jo

**Comment by KevinJump** [comment]
Hi, 

I think github's email filter stripped out most of the code :( - can you update it on the issue and i will take a look

**Comment by steadygo** [comment]
> Hi,
> 
> I think github's email filter stripped out most of the code :( - can you update it on the issue and i will take a look

Hi Kevin

I have edited and attached the relevant screens above.

**Comment by KevinJump** [comment]
Hi, 

I think we have found the culprit from a uSync point of view, we don't set multi-url picker values to null but rather blank, when inside nested content or blocks. this is an issue when the Multi-Url picker is set to a max of one item and nothing is picked. 

this is something we can put a fix in for - we just need to do some testing 

it does also look like the internal error is fixed in umbraco too. from v9.5 onwards. 

https://github.com/umbraco/Umbraco-CMS/commit/6eebd9660530076a27a0f442edd6c35cad964499

**Comment by steadygo** [comment]
Hi Kevin

OK. Good to know – I’ll update Umbraco when I can and await a Usync update.

Really appreciate your input.

Many Thanks

Jo

**Comment by KevinJump** [comment]
## For our own records. 

This issue is fixed in multiple versions of Umbraco -

- [x] Fix in v8.18.0 - (https://github.com/umbraco/Umbraco-CMS/commit/e8c18cda56b2b8a15692ff718af7b1edc94e87af)
- [x] Fix in v9.5 (https://github.com/umbraco/Umbraco-CMS/pull/12109)
- [x] v10 not effected 

As there is fix in all the current versions of Umbraco we will release the uSync 'fix' as part of a regular patch 

- [x] v8 Fix (https://github.com/KevinJump/uSync/commit/ed0fb8d9a0e0f177fbb837bf30a5ca27900f14f6)
- [ ] v9 Fix 
- [x]  v10 'fix' (https://github.com/KevinJump/uSync/pull/401)

**Comment by KevinJump** [comment]
Tidy up - this is fixed in all supported releases.
